cryptoki/src/mechanism/aead.rs

```diff
@@ -53,23 +49,35 @@ impl<'a> GcmParams<'a> {
         // set it to zero.
         //
         // [1]: https://www.oasis-open.org/committees/document.php?document_id=58032&wg_abbrev=pkcs11
-        GcmParams {
+
+        let iv_len = iv.len();
+        // Some HSMs may require the ulIvBits field to be populated, while others don't pay attention to it.
+        let iv_bit_len = iv_len*8;
+
+        Ok(GcmParams {
             inner: CK_GCM_PARAMS {
                 pIv: iv.as_mut_ptr(),
-                ulIvLen: iv
-                    .len()
-                    .try_into()
-                    .expect("iv length does not fit in CK_ULONG"),
-                ulIvBits: 0,
+                ulIvLen: match iv_len.try_into() {
+                    Ok(len) => len,
+                    Err(_e) => return Err("iv length does not fit in CK_ULONG"),
+                },
+                // Since this field isn't universally used, set it to 0 if it doesn't fit in CK_ULONG.
+                // If the HSM doesn't require the field, it won't mind; and it it does, it would break anyways.
+                ulIvBits: match iv_bit_len.try_into() {
+                    Ok(len) => len,
+                    Err(_e) => 0,
+                },
```

Comment:
```suggestion
                ulIvBits: iv_bit_len.try_into().unwrap_or_default(),
```

Would that work? :thinking: